app/models.py

```diff
@@ -124,56 +139,90 @@ def create_from_images(images, project):
         return None
 
     def process(self):
-        # Nothing to do if we don't have a processing node...
-        if not self.processing_node: return
-        
-        # Need to process some images (UUID not yet set)?
-        if not self.uuid:
-            print("Processing... {}".format(self))
+        """
+        This method contains the logic for processing tasks asynchronously
+        from a background thread or from the scheduler. Here tasks that are
+        ready to be processed execute some logic. This could be communication
+        with a processing node or executing a pending action.
+        """
+
+        if self.processing_node:
+            # Need to process some images (UUID not yet set)?
+            if not self.uuid:
+                logger.info("Processing... {}".format(self))
+
+                images = [image.path() for image in self.imageupload_set.all()]
```

Comment:
Where / how are images (and intermediate / completed artifacts) stored so that processing nodes have access to them?